build.rs

```diff
@@ -13,3 +17,55 @@ fn main() -> Result<(), Box<dyn Error>> {
 
     Ok(())
 }
+
+fn gen_presets_hook(mut file: &File) -> SdResult<()> {
+    println!("cargo:rerun-if-changed=docs/.vuepress/public/presets/toml");
+    let paths = fs::read_dir("docs/.vuepress/public/presets/toml")?;
+
+    let mut presets = String::new();
+    let mut match_arms = String::new();
+    for path in paths {
+        let unwrapped = path?;
+        let file_name = unwrapped.file_name();
+        let full_path = dunce::canonicalize(unwrapped.path())?;
+        let full_path = full_path.to_str().expect("failed to convert to string");
+        let name = file_name
+            .to_str()
+            .and_then(|v| v.strip_suffix(".toml"))
+            .expect("Failed to process filename");
+        presets.push_str(format!("print::Preset(\"{}\"),\n", name).as_str());
+        match_arms.push_str(
+            format!(
+                r#"
+"{name}" => {{
+        let mut stdout = io::stdout().lock();
+        stdout.write_all(include_bytes!(r"{full_path}")).expect("Failed to print preset");
```

Comment:
`expect` makes for poor error messages (the other instances in this PR should never be hit, so they should be fine). I think doing nothing is preferable here, as printing an error message when something like `head` is used is unexpected. (#885)

```suggestion
        let _ = stdout.write_all(include_bytes!(r"{full_path}"));
```